src/query/interpret/eval.hpp

```diff
@@ -479,9 +479,37 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       case TypedValue::Type::Null:
         return TypedValue(ctx_->memory);
       case TypedValue::Type::Vertex:
-        return TypedValue(GetProperty(expression_result_ptr->ValueVertex(), property_lookup.property_), ctx_->memory);
+        if (property_lookup.evaluation_mode_ == PropertyLookup::EvaluationMode::GET_ALL_PROPERTIES) {
+          auto symbol_pos = static_cast<Identifier *>(property_lookup.expression_)->symbol_pos_;
+          if (!ctx_->property_lookups_cache.contains(symbol_pos)) {
+            ctx_->property_lookups_cache[symbol_pos] = GetAllProperties(expression_result_ptr->ValueVertex());
+          }
+
+          auto property_id = ctx_->properties[property_lookup.property_.ix];
+          if (!ctx_->property_lookups_cache.at(symbol_pos).contains(property_id)) {
+            return TypedValue(ctx_->memory);
+          }
+
+          return TypedValue(ctx_->property_lookups_cache.at(symbol_pos).at(property_id), ctx_->memory);
```

Comment:
See [comment](https://github.com/memgraph/memgraph/pull/1168#discussion_r1300668664) about null properties

Reply:
Here I was focusing on the try catch block. The throw was there just from my general comments.
`at(...)` checks is the key exists and throws if not found. So instead of having `if (x.contains(...)) x.at(...)` you could do one of two things.
1) `if (x.contains(...)) x[...]`, [] doesn't check, but we already checked it
2) `try {x.at(...)} catch(std::out_of_range) {...}`, at will check and throw, so we could skip the if

Since we are returning the `null` object, I think the first solution should work better.

Reply:
Switched to your first solution